File: solutions/0122-Best-Time-to-Buy-and-Sell-Stock-II -Solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = function(prices) {
  let acc = 0;
  
  for (let i = 0; i < prices.length; i++) {
    if (prices[i] < prices[i + 1]) acc += prices[i + 1] - prices[i];
  }
  return acc;
};

File: solutions/0122-Best-Time-to-Buy-and-Sell-Stock-II-Solution.cpp
class Solution {
public:
  int maxProfit(vector<int>& prices) {
    bool buy = true;
    int transaction {0};
    int profit {0};
    
    for (auto it = prices.begin(); it != prices.end(); ++it) {      
      if(!buy and (it + 1) == prices.end()) {
        profit += *it - transaction;
        break;
      }
      
      if(buy and (it + 1) == prices.end()) break;
      
      if (!buy and (*it > *(it + 1))) {
        buy = true;
        profit += *it - transaction;
        continue;
      }
      
      if (buy and (*it < *(it + 1))) {
        buy = false;
        transaction = *it;
      }
    }
    
    return profit;
  }
};
